scroll-search.lic

```diff
@@ -0,0 +1,520 @@
+        return
+      end
+
+      echo("Scroll scout is: #{@owned_scrolls.length}") if $debug_mode
+      display_scrolls()
+    else
+      echo("***INVALID ACTION***")
+    end
+  end
+
+  def search_container(container)
+    # return if invalid container
+
+	unless exists?(container)
+      echo("*** INVALID CONTAINER. ***")
+      return
+    end
+
+    # Get long name of container
+    $container_full = full_name(container)
+    $container = get_noun($container_full)
+
+    # iterate over all scroll types
+    scroll_types = $settings.scroll_nouns
+
+    if scroll_types.length <= 0 || scroll_types.nil?
+      echo("*** NO SCROLL TYPTES SET! SETUP YOUR YAML TO INCLUDE VALID \"scroll_types\". ***")
+      return
+    end
+
+    echo("Scroll type count is: #{scroll_types.length}.") if $debug_mode
+
+	# Clear current scroll listing from owned scrolls if
+	# user is searching an existing container
+	if @owned_scrolls.key? ($container_full)
+		echo("Clearing scroll array for this container, since it was previously searched --- #{$container_full}!") if $debug_mode
+		@owned_scrolls[$container_full] = []
+	end
+	
+	# reset scroll count
+	@scroll_count = 0
+	
+    scroll_types.each do |scroll|
+      echo "Searching for scrolls of type #{scroll}." if $debug_mode
+
+      scroll_list = get_scrolls(scroll)
+      next if scroll_list.empty?
+
+      look_scrolls(scroll_list)
+    end
+
+    # Save scrolls to user variables.
+	if UserVars.owned_scrolls.nil? && @owned_scrolls.nil?
+		echo("Creating an empty hash for vars!") if $debug_mode
+		UserVars.owned_scrolls = {}
+		return
+	end
+	
+	existing_hash = if UserVars.owned_scrolls.nil? then @owned_scrolls.clone else UserVars.owned_scrolls.clone end
+	
+	new_hash = @owned_scrolls
+	existing_hash.each do |container, scroll_array|
+		if new_hash.key? (container)
+			echo("Existing hash already has this container: #{container}") if $debug_mode
+			next
+		else
+			new_hash[container] = scroll_array
+		end
+	end
+	
+    UserVars.owned_scrolls = new_hash
+	UserVars.scroll_counter = @global_scroll_index
+	
+    echo ("Total scrolls added: #{@owned_scrolls.length}!") if $debug_mode
+	
+	echo ("*** SEARCH COMPLETED ***")
+  end
+
+  def display_scrolls()
+    current_scrolls = UserVars.owned_scrolls
+
+    if current_scrolls.length <= 0 || current_scrolls.nil?
+      echo ("*** NO SCROLLS TO DISPLAY -- SEARCH FIRST! ***")
+      return
+    end
+
+	
+    echo ("|============================== Scroll List =============================|")
+    echo (Kernel::sprintf("|%s%s%s%s%s%s%s   %s", format_bar(3, "#"), format_bar(3, "|"), format_bar(14, "Scroll Name"), format_bar(7, "|"),
+																	  format_bar(current_scrolls[0].get_container.length - "Container".length, "Container"), format_bar("Container|".length, "|"), 
+																	  format_bar(5 + "Type".length, "Type"), format_bar("Type".length, "|")))
+    echo ("|========================================================================|")
```

Comment:
We can move away from sprintf without losing much here, and it would be a lot simpler in a single helper method that you could use for the header and the body of the table.

```
def echo_scroll_list_line(index, name, container)
  echo("|#{index.to_s.center(6)}|#{name.center(30)}|#{container.center(15)}|")
end
```